### PR TITLE
Create unified GitHub Actions CI/CD workflow

### PR DESCRIPTION
**Pull request by madig**
Which passes files around as needed instead of rebuilding fonts for each step.

---

@chrissimpkins edits below

### TODO

- [x] test git commit tag push => release build workflow

### CODE REVIEW

**Comment by madig** [comment]
According to https://github.com/googlefonts/googlesans/actions/runs/2557773318/usage, they actually round up to the full minute for billing. Maybe the lining jobs should be combined, at least ufolint and kernlint.

**Comment by madig** [comment]
(Rerun succeeded at https://github.com/googlefonts/googlesans/actions/runs/2733193914)

**Comment by chrissimpkins** [comment]
@madig Will you please update the maintainer documentation with the process to 

- build and CI test variable fonts
- build and CI test static fonts (seems to be triggered through a separate process, similar to our current approach)
- push a branch that uploads static and variable artifacts to the GH file server for testing
- push a new production GitHub release and release archive

with the changes here? Are you proposing a change in any of our current processes?

**Comment by chrissimpkins** [comment]
I'll test the git commit tag push to execute the updated release workflow

**Comment by chrissimpkins** [comment]
> @madig Will you please update the maintainer documentation with the process to
> 
> * build and CI test variable fonts
> * build and CI test static fonts (seems to be triggered through a separate process, similar to our current approach)
> * push a branch that uploads static and variable artifacts to the GH file server for testing
> * push a new production GitHub release and release archive
> 
> with the changes here? Are you proposing a change in any of our current processes?

@madig Let me know when you have a chance

**Comment by chrissimpkins** [comment]
Triggered the updates here with new version git commit tag

https://github.com/googlefonts/googlesans/actions/runs/2760676731

**Comment by chrissimpkins** [comment]
The git tag push appropriately triggered a release.

File: .github/workflows/cicd.yml [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: Google Sans ${{ github.ref }}
+          body: |
+            Please see the root of the repository for the CHANGELOG.md
+          draft: false
```
Comment (chrissimpkins) [inline .github/workflows/cicd.yml]: Mind changing `draft` to `true`?  This is a change from our previous approach. I think you set it this way in a previous round of Actions work (or maybe on GS Thai Loopless?) and it is nice to be able to review the draft automated release before it is an official release.

**Review by chrissimpkins — CHANGES_REQUESTED** [review]

**Comment by madig** [comment]
(Will respond later today hopefully)

**Comment by chrissimpkins** [comment]
> (Will respond later today hopefully)

No rush as long as you aren't blocked on landing this.  Thanks Nikolaus!

**Pull request merged**